gnd/src/main/java/com/google/android/gnd/ui/field/FieldView.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.field;
+
+import android.view.LayoutInflater;
+import android.widget.FrameLayout;
+import androidx.lifecycle.LifecycleOwner;
+import com.google.android.gnd.model.form.Field;
+import com.google.android.gnd.model.observation.Response;
+import com.google.android.gnd.ui.common.AbstractFragment;
+import com.google.android.gnd.ui.editobservation.EditObservationFragmentArgs;
+import com.google.android.gnd.ui.util.ViewUtil;
+import java8.util.Optional;
+
+public abstract class FieldView extends FrameLayout {
```

Comment:
Do we actually need a custom class for the View at all? View classes are harder to tests since they're tied to the Android UI APIs and lifecycle. Instead, could the factory be used to inflate the relevant layout, and to create the appropriate ViewModel?